gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/dataset/ConvertibleHiveDataset.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.data.management.conversion.hive.dataset;
+
+import java.util.HashSet;
+import java.util.Properties;
+
+import lombok.Getter;
+import lombok.ToString;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.hive.ql.metadata.Table;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Sets;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigException;
+import com.typesafe.config.ConfigFactory;
+
+import gobblin.data.management.copy.hive.HiveDataset;
+import gobblin.hive.HiveMetastoreClientPool;
+import gobblin.util.ConfigUtils;
+
+/**
+ * A {@link HiveDataset} that can be converted to another {@link HiveDataset}
+ */
+@Getter
+@ToString
+public class ConvertibleHiveDataset extends HiveDataset {
+
+  public static final String DESTINATION_TABLE_KEY = "destination.tableName";
+  public static final String DESTINATION_DB_KEY = "destination.dbName";
+  public static final String DESTINATION_DATA_PATH_KEY = "destination.dataPath";
+  public static final String DESTINATION_CONVERSION_FORMATS_KEY = "destination.formats";
+
+  private static final String DB_NAME_TEMPLATE = "{dbName}";
```

Comment:
the hive copier uses the templates `$DB` and `$TABLE`, can you use those for consistency?